### PR TITLE
[#118148757] Updated WordBoundary logic considering with adjacent words

### PR DESCRIPTION
**Pull request by toshipon**
### Problem

if we have `test` content token which want to highlight in `This is 2test`. 
it doesn't allow to highlight because of cleanup boundary function behavior now. 
`new RegExp('\\b' + this.escapeRegExp(w) + '\\b').test(ww);`
### Solution

I'd like to accept these cases as tokens: 
- highlight `test` content in  `This is 2test` (ignore adjacent digit)
- highlight `2` content in `This is 2test` (ignore adjacent non-digit)

so that I've changed a wordboundary logic to consider adjacent words

### CODE REVIEW

File: spec/javascripts/helper.spec.js [hunk 1]
```diff
@@ -335,6 +335,35 @@ describe('Helper', function() {
...
+        { 'class': 'test', 'match': ['Text', 'test'] }
+      ];
+
+      var _indecies = [
+        { start: 0, end: 4, type: 'test' },
+        { start: 6, end: 10, type: 'test' },
+        { start: 14, end: 18, type: 'test' },
+      ]
+
+      var indecies = generateIndecies(text, list);
```
Comment (kelvintaywl) [inline spec/javascripts/helper.spec.js]: [spelling / format]
i suppose we meant:

``` diff
- indecies
+ indices
```

**Comment by kelvintaywl** [comment]
would we need to care about special characters?
for example, `@david` should be considered a word too? else, i guess we can scoping this to only cater to numbers?

**Comment by kelvintaywl** [comment]
LGTM besides the spelling issue! 🙇

File: spec/javascripts/helper.spec.js (hunk 1 above)
Comment (toshipon) [inline spec/javascripts/helper.spec.js]: thanks ! fixed it 🙇

**Comment by kelvintaywl** [comment]
LGTM

**Comment by mikamikuh** [comment]
LGTM 👍

**Pull request merged**